eng/pipelines/global-build.yml

```diff
@@ -81,6 +81,25 @@ extends:
                 eq(stageDependencies.EvaluatePaths.evaluate_paths.outputs['SetPathVars_non_wasm.containsChange'], true),
                 eq(variables['isRollingBuild'], true))
 
+      # Re-enable when fixing https://github.com/dotnet/runtime/issues/108628
```

Comment:
Would it make more sense to build the `TRACE_GC` instrumentation into the checked builds by default and have a config switch it to enable/disable it? It is what we do everywhere else.

Adding a new CI leg to validate a few dozens of ifdefs is not very scalable.

Reply:
I'm still sorting through the details on this.  The GC appears to have some STRESS_LOG instrumentation that is already available by default (normal stress log).  TRACE_GC (without SIMPLE_DPRINTF) allows GC dprintfs to be added to the stress log, but it abuses the LogFacility value:
``` c++
    static void LogMsg(unsigned dprintfLevel, const StressLogMsg& msg)
    {
        GCToEEInterface::LogStressMsg(LL_ALWAYS, LF_ALWAYS|(dprintfLevel<<16)|LF_GC, msg);
    }
```
in a way that seems like it would be undesirable for others.  LogStressAnalyzer filters on this.  (I don't know the history, but it looks like the facility might have previously been 16 bits because I feel like I've seen various assumptions about that as I poke around.)

Reply:
It turns out that the "available by default" instrumentation is for things like roots.  The internal GC instrumentation is all under TRACE_GC.  Some of that instrumentation at a very low level (used for tracking movement within the heap) that is believed to have substantial overhead (even if off).

It looks like TRACE_GC and DOTNET_StressLog (if not restricted by DOTNET_LogFacility) can collide, though the existence of the low bit (LF_GC) can be used to disambiguate (even though it otherwise seems to be intended as a pure bitmask).  Interestingly, 

I'm contemplating putting in a pre-STRESS_LOG_VA condition.  In fact, the checked-in code already has a "commonly used line" -
``` C++
#define dprintf(l,x) STRESS_LOG_VA(l,x);
//#define dprintf(l,x) {if ((l <= 2) || (l == 6666)) {STRESS_LOG_VA(l,x);}}
```
Using
``` C++
#define dprintf(l,x) {if (l == *1*) {STRESS_LOG_VA(*0*,x);}}
```
is subtle but might work (need to verify), and then developers could opt-in as they please.  Having a static check like this would mitigate the cost in checked/release builds.

Reply:
Please see #108655

Reply:
As your PR has all of the fixes, would you feel comfortable if we close my PR and merge yours instead?

Reply:
Sure - happy to either include it or deal with the merge.  The only question would be if mine has grown too big or if the GC changes need more discussion and delay it.

Reply:
I think it makes more sense to do the whole "Fix the TRACE_GC" build work in one PR. I'll close this one.